pal/src/map/virtual.cpp

```diff
@@ -878,15 +870,14 @@ static LPVOID VIRTUALReserveMemory(
     MemSize = ( ((UINT_PTR)lpAddress + dwSize + VIRTUAL_PAGE_MASK) & ~VIRTUAL_PAGE_MASK ) -
                StartBoundary;
 
-    InternalEnterCriticalSection(pthrCurrent, &virtual_critsec);
-
-    // If this is a request for special executable (JIT'ed) memory then, first of all,
-    // try to get memory from the executable memory allocator to satisfy the request.
-    if (((flAllocationType & MEM_RESERVE_EXECUTABLE) != 0) && (lpAddress == NULL))
+    if ((flAllocationType & MEM_RESERVE_EXECUTABLE) != 0)
     {
-        pRetVal = g_executableMemoryAllocator.AllocateMemory(MemSize);
+        fprintf(stderr, "MEM_RESERVE_EXECUTABLE is not supported!");
```

Comment:
@curtisman also asked for keeping CLR related stuff that I removed with one of the commits on this PR. ATM, it just makes it more complicated and error prone. IMHO once we want to enable it, we may better approach VirtualAlloc/Free stuff similar to other PlatformAgnostic items. i.e. On some of the bench runs, I saw whole run took ~10 secs while the total time we spent on VirtualFree on xplat was ~1.52 secs.. I want to make some serious changes here but not pre-release indeed.